pkg/controller/csi_handler_test.go

```diff
@@ -511,27 +517,27 @@ func TestCSIHandler(t *testing.T) {
 		{
 			name:           "Node without annotations -> error",
 			initialObjects: []runtime.Object{pvWithFinalizer(), nodeWithoutAnnotations()},
-			addedVA:        va(false, fin),
+			addedVA:        va(false, fin, ann),
 			expectedActions: []core.Action{
-				core.NewUpdateAction(vaGroupResourceVersion, metav1.NamespaceNone, vaWithAttachError(va(false, fin), "node \"node1\" has no NodeID annotation")),
+				core.NewUpdateAction(vaGroupResourceVersion, metav1.NamespaceNone, vaWithAttachError(va(false, fin, ann), "node \"node1\" has no NodeID annotation")),
 			},
 		},
 		{
 			name:           "CSINodeInfo exists without the driver, Node without annotations -> error",
```

Comment:
wonder if this should work now cause we get the nodeid from the VA

Reply:
That's a good question. Something has removed the annotation from the node. VA annotation is probably the best option here. I'll fix it.

Reply:
Oh, sorry, this is about attach. In order to attach, a node **must** have annotation with NodeID (or CSINodeInfo instance). I think that node without Node ID is fundamentally broken and attach should fail. At least, the driver is not registered to the node.